We will use CatBoost as base model. At first, let's try to put data to the model without transformations. After that we will try to replace missing values by -1. After that we will try to fill gaps with KNNImputer from Scikit Learn. Next idea to try is filling the missing values with average values depending on class labels.

In [1]:
import pandas as pd 
import numpy as np
import seaborn as sns
import missingno as msno
from matplotlib import pyplot as plt
from sklearn import preprocessing
from sklearn.model_selection import train_test_split

In [3]:
from numpy.lib.function_base import average
from sklearn.pipeline import Pipeline
from sklearn.impute import KNNImputer, SimpleImputer
# from imblearn.over_sampling import SMOTE

from sklearn.model_selection import cross_val_score
from sklearn.model_selection import StratifiedKFold
from tqdm import tqdm

In [78]:
pip install catboost

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [4]:
from catboost import CatBoostClassifier

In [79]:
#pip install imbalanced-learn
#pip install missingno
#pip install catboost

In [5]:
from matplotlib import rcParams

# figure size in inches
rcParams['figure.figsize'] = 15,12

In [ ]:
from google.colab import drive
drive.mount('drive')

Mounted at drive


In [6]:
data = pd.read_csv('../CSV/Norwey data.csv', sep=';')
# data

In [7]:
features = data.loc[:, "CALI":"RXO"]
# features

In [8]:
targets = data['FORCE_2020_LITHOFACIES_LITHOLOGY']

label_encoder = preprocessing.LabelEncoder()
encoded_targets = label_encoder.fit_transform(targets)

In [9]:
gaps_pct_threshold = 50

# Percent of missing values in data
nans_pct = (features.isnull().sum()/features.shape[0]*100).astype('int').sort_values(ascending=False)
# We select features with missing values less than n %
good_columns = nans_pct[nans_pct < gaps_pct_threshold].index.to_list()
good_features = features[good_columns]
good_targets = encoded_targets[good_features.index]

In [11]:
def fill_nans_with_value_from_distribution(data, target):
    data_n_target = data.join(target)
    target_col_name = target.name
    # for class_name in np.unique(target):
    for col_name in data.columns:
        data_n_target[col_name] = data_n_target[col_name].fillna(data_n_target.groupby(target_col_name)[col_name].transform('mean'))
    return data_n_target.drop(columns=[target_col_name])

**1. CatBoost and Virgin Data**

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(features, encoded_targets, test_size=0.3, random_state=1, shuffle=True, stratify=encoded_targets)

catboost_model = CatBoostClassifier(iterations=70,
                           learning_rate=0.5,
                           depth=2,
                           custom_metric=['TotalF1:average=Macro']
                           )

catboost_model.fit(X_train, y_train, verbose=False)

score_1 = catboost_model.score(X_test, y_test)

print(f"Score: {score_1}")

Score: 0.7648723921698173


**2. CatBoost + normalized data and replaced missing values**

In [ ]:
scaler = preprocessing.StandardScaler()

imputer = SimpleImputer(strategy='constant', fill_value=-1)

X_train, X_test, y_train, y_test = train_test_split(features, encoded_targets, random_state=0)

catboost_model = CatBoostClassifier(iterations=70,
                           learning_rate=0.5,
                           depth=2,
                           custom_metric=['TotalF1:average=Macro']
                           )

pipeline_2 = Pipeline(steps=[('scaler', scaler), ('imputer', imputer), ('model', catboost_model)])

pipeline_2.fit(X_train, y_train)

score_2 = pipeline_2.score(X_test, y_test)

print(f"Score: {score_2}")

0:	learn: 1.2678461	total: 1.56s	remaining: 1m 47s
1:	learn: 1.1104466	total: 2.81s	remaining: 1m 35s
2:	learn: 1.0550017	total: 4s	remaining: 1m 29s
3:	learn: 0.9970682	total: 5.34s	remaining: 1m 28s
4:	learn: 0.9616488	total: 6.73s	remaining: 1m 27s
5:	learn: 0.9410390	total: 7.99s	remaining: 1m 25s
6:	learn: 0.9259078	total: 9.26s	remaining: 1m 23s
7:	learn: 0.9116644	total: 10.6s	remaining: 1m 21s
8:	learn: 0.9120892	total: 11.7s	remaining: 1m 19s
9:	learn: 0.8929877	total: 13.8s	remaining: 1m 22s
10:	learn: 0.8782309	total: 15.4s	remaining: 1m 22s
11:	learn: 0.8694416	total: 16.8s	remaining: 1m 21s
12:	learn: 0.8624586	total: 18.2s	remaining: 1m 19s
13:	learn: 0.8527893	total: 19.7s	remaining: 1m 18s
14:	learn: 0.8716393	total: 20.8s	remaining: 1m 16s
15:	learn: 0.8491921	total: 21.9s	remaining: 1m 13s
16:	learn: 0.8407420	total: 23.6s	remaining: 1m 13s
17:	learn: 0.8344396	total: 24.8s	remaining: 1m 11s
18:	learn: 0.8307386	total: 26s	remaining: 1m 9s
19:	learn: 0.8231496	total: 

We did not get significantly high score from both ideas. Let's move further

**3. CatBoost + Features that have less than 50% of missing values**

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(good_features, good_targets, test_size=0.3, random_state=1, shuffle=True, stratify=encoded_targets)

catboost_model = CatBoostClassifier(iterations=70,
                           learning_rate=0.5,
                           depth=2,
                           custom_metric=['TotalF1:average=Macro']
                           )

catboost_model.fit(X_train, y_train, verbose=False)

score_3 = catboost_model.score(X_test, y_test)

print(f"Score: {score_3}")

Score: 0.7630840030300097


Removing of features with huge number of missing values decreases the score

**4. CatBoost and Stochastic KNNImputer**

In [ ]:
def stochastic_knn_imputer(knn_imputer, data, num_butches, fit_or_transform):
    data_after_knn_processing = None
    previous_chunk = None
    for chunk in tqdm(np.array_split(data, num_butches)):   # 24579 rows in chunk for X_train
        if previous_chunk is not None:
            chunk = np.concatenate((previous_chunk, chunk))
            previous_chunk = None
        if ((np.isnan(chunk).sum(axis=0)/chunk.shape[0]*100)==100).any():   # if any column in chunk has 100% nans then we concatenate it with next cbunk 
            previous_chunk = chunk
            continue
        imputted_chunk = knn_imputer.fit_transform(chunk) if fit_or_transform == 'fit' else knn_imputer.transform(chunk)
        if data_after_knn_processing is None:
            data_after_knn_processing = imputted_chunk
        else:
            data_after_knn_processing = np.concatenate((data_after_knn_processing, imputted_chunk))
    return data_after_knn_processing

In [ ]:
X_train, X_test, y_train, y_test = train_test_split(good_features, good_targets, test_size=0.3, random_state=1, shuffle=False)

imputer = KNNImputer(n_neighbors=3, weights='uniform', metric='nan_euclidean', missing_values=np.nan)

# fit stochastic knn imputer
X_train_imputted = stochastic_knn_imputer(imputer, X_train, 100, 'fit')
X_test_impitted = stochastic_knn_imputer(imputer, X_test, 14, 'transform')

catboost_model = CatBoostClassifier(iterations=70,
                           learning_rate=0.5,
                           depth=2,
                           custom_metric=['TotalF1:average=Macro']
                           )

catboost_model.fit(X_train_imputted, y_train)

score_4 = catboost_model.score(X_test_impitted, y_test)
print(f"Score: {score_4}")

  0%|          | 0/10 [00:00<?, ?it/s]

**5. CatBoost + Filling missing values with average values depending on a class label**

In [12]:
X_train, X_test, y_train, y_test = train_test_split(good_features, good_targets, test_size=0.3, random_state=1, shuffle=True, stratify=encoded_targets)

prepared_train_target = pd.Series(y_train, index=X_train.index, name=targets.name)
prepared_test_target = pd.Series(y_test, index=X_test.index, name=targets.name)

X_train_filled = fill_nans_with_value_from_distribution(X_train, prepared_train_target)
X_train_filled = X_train_filled.fillna(X_train_filled.mean())
y_train_prepared = prepared_train_target[X_train_filled.index]

X_test_filled = fill_nans_with_value_from_distribution(X_test, prepared_test_target)
X_test_filled = X_test_filled.fillna(X_test_filled.mean())
y_test_prepared = prepared_test_target[X_test_filled.index]

catboost_model = CatBoostClassifier(iterations=70,
                           learning_rate=0.5,
                           depth=2,
                           custom_metric=['TotalF1:average=Macro']
                           )

catboost_model.fit(X_train_filled, y_train_prepared, verbose=False)

score_5 = catboost_model.score(X_test_filled, y_test_prepared)

print(f"Score: {score_5}")

Score: 0.9092762719490594


Filling NANs with the average value depending on class label helps to improve data significantly. Score rises from .77 to .90

**6. Data Oversampling** <br>
We have minority class with 103 samples. Let us oversample data to help the model to get more info from data

In [ ]:
pd.Series(good_targets).value_counts()

1     720803
0     168937
2     150455
3      56320
6      33329
11     15245
4      10513
8       8213
9       3820
5       1688
7       1085
10       103
dtype: int64

In [ ]:
strategy = {10:400}  # increasing the minority class size
oversample = SMOTE(sampling_strategy=strategy)
X_train_oversampled, y_train_oversampled = oversample.fit_resample(X_train_filled, y_train_prepared)

catboost_model = CatBoostClassifier(iterations=70,
                           learning_rate=0.5,
                           depth=2,
                           custom_metric=['TotalF1:average=Macro']
                           )

catboost_model.fit(X_train_oversampled, y_train_oversampled, verbose=False)

score_6 = catboost_model.score(X_test_filled, y_test_prepared)

print(f"Score: {score_6}")

Score: 0.9044863507179186


In [ ]:
pd.Series(y_train_oversampled).value_counts()

1     504562
0     118256
2     105318
3      39424
6      23330
11     10671
4       7359
8       5749
9       2674
5       1182
7        760
10       300
Name: FORCE_2020_LITHOFACIES_LITHOLOGY, dtype: int64

Oversampling method did not help us to improve the data.

**Model tuning**

In [13]:
pip install scikit-optimize

Note: you may need to restart the kernel to use updated packages.


In [37]:
from skopt import gp_minimize
from skopt.space import Real, Integer, Categorical
from skopt.utils import use_named_args

data = (X_train_filled, y_train_prepared, X_test_filled, y_test_prepared)

space = [Integer(100, 500, name='iterations'),
         Real(0.01, 1, 'log-uniform', name='learning_rate'),
         Integer(2, 10, name='depth'),
         Integer(1, 10, name='l2_leaf_reg'),
         # Categorical(['TotalF1:average=Macro'], name='custom_metric'),
        ]

@use_named_args(space)
def objective(**params):
    X_train, y_train, X_test, y_test = data
    model = CatBoostClassifier(**params, custom_metric=['TotalF1:average=Macro'])
    model.fit(X_train, y_train)
    return -model.score(X_test, y_test)

result = gp_minimize(objective, space, n_calls=20, random_state=1, verbose=True, n_jobs=-2)
print('Best Score: %.3f' % (-result.fun))
print('Best Parameters:' + (" {:.5f}" * len(result.x)).format(*result.x))


Iteration No: 1 started. Evaluating function at random point.
0:	learn: 1.1549213	total: 399ms	remaining: 3m 18s
1:	learn: 8.4807190	total: 759ms	remaining: 3m 8s
2:	learn: 1009.5566510	total: 1.19s	remaining: 3m 16s
3:	learn: 765.0088340	total: 1.59s	remaining: 3m 17s
4:	learn: 765.8676411	total: 1.95s	remaining: 3m 12s
5:	learn: 705.4610237	total: 2.35s	remaining: 3m 12s
6:	learn: 1690.8897420	total: 2.73s	remaining: 3m 11s
7:	learn: 1526.6028286	total: 3.21s	remaining: 3m 17s
8:	learn: 1224.3346404	total: 3.63s	remaining: 3m 17s
9:	learn: 1458.7905626	total: 4.05s	remaining: 3m 18s
10:	learn: 1143.3737648	total: 4.46s	remaining: 3m 17s
11:	learn: 1078.8347870	total: 4.9s	remaining: 3m 18s
12:	learn: 1052.0748367	total: 5.34s	remaining: 3m 19s
13:	learn: 1005.6367855	total: 5.77s	remaining: 3m 19s
14:	learn: 957.6038069	total: 6.35s	remaining: 3m 24s
15:	learn: 1223.5934772	total: 6.82s	remaining: 3m 25s
16:	learn: 1199.6311109	total: 7.3s	remaining: 3m 27s
17:	learn: 1109.8417600	to

151:	learn: 192.2269063	total: 1m 4s	remaining: 2m 28s
152:	learn: 190.1199764	total: 1m 5s	remaining: 2m 27s
153:	learn: 189.5214387	total: 1m 5s	remaining: 2m 27s
154:	learn: 189.0971259	total: 1m 6s	remaining: 2m 26s
155:	learn: 188.0702277	total: 1m 6s	remaining: 2m 26s
156:	learn: 187.0423922	total: 1m 6s	remaining: 2m 25s
157:	learn: 186.4449128	total: 1m 7s	remaining: 2m 25s
158:	learn: 186.1710443	total: 1m 7s	remaining: 2m 24s
159:	learn: 184.9733594	total: 1m 8s	remaining: 2m 24s
160:	learn: 184.8326064	total: 1m 8s	remaining: 2m 23s
161:	learn: 183.9927676	total: 1m 8s	remaining: 2m 23s
162:	learn: 183.1825332	total: 1m 9s	remaining: 2m 22s
163:	learn: 182.7297208	total: 1m 9s	remaining: 2m 22s
164:	learn: 181.8326345	total: 1m 10s	remaining: 2m 22s
165:	learn: 181.4866552	total: 1m 10s	remaining: 2m 21s
166:	learn: 180.0211848	total: 1m 11s	remaining: 2m 21s
167:	learn: 179.4902251	total: 1m 11s	remaining: 2m 20s
168:	learn: 178.1268328	total: 1m 11s	remaining: 2m 20s
169:	

299:	learn: 129.3791890	total: 2m 5s	remaining: 1m 23s
300:	learn: 129.1358236	total: 2m 5s	remaining: 1m 22s
301:	learn: 128.9739922	total: 2m 6s	remaining: 1m 22s
302:	learn: 128.7103338	total: 2m 6s	remaining: 1m 21s
303:	learn: 128.6750667	total: 2m 6s	remaining: 1m 21s
304:	learn: 128.2856266	total: 2m 7s	remaining: 1m 21s
305:	learn: 128.1587864	total: 2m 7s	remaining: 1m 20s
306:	learn: 127.9923520	total: 2m 8s	remaining: 1m 20s
307:	learn: 127.8158914	total: 2m 8s	remaining: 1m 19s
308:	learn: 127.5697877	total: 2m 9s	remaining: 1m 19s
309:	learn: 127.3199948	total: 2m 9s	remaining: 1m 18s
310:	learn: 127.0352053	total: 2m 9s	remaining: 1m 18s
311:	learn: 126.8717474	total: 2m 10s	remaining: 1m 18s
312:	learn: 126.7104884	total: 2m 10s	remaining: 1m 17s
313:	learn: 126.5718335	total: 2m 10s	remaining: 1m 17s
314:	learn: 126.4124688	total: 2m 11s	remaining: 1m 16s
315:	learn: 126.3969472	total: 2m 11s	remaining: 1m 16s
316:	learn: 126.2340119	total: 2m 12s	remaining: 1m 15s
317:

449:	learn: 107.6351552	total: 3m 6s	remaining: 20.3s
450:	learn: 107.5989331	total: 3m 6s	remaining: 19.8s
451:	learn: 107.5809528	total: 3m 6s	remaining: 19.4s
452:	learn: 107.0823759	total: 3m 7s	remaining: 19s
453:	learn: 107.0162956	total: 3m 7s	remaining: 18.6s
454:	learn: 106.8157204	total: 3m 8s	remaining: 18.2s
455:	learn: 106.1132323	total: 3m 8s	remaining: 17.8s
456:	learn: 105.9216030	total: 3m 8s	remaining: 17.4s
457:	learn: 105.6447323	total: 3m 9s	remaining: 16.9s
458:	learn: 105.3044195	total: 3m 9s	remaining: 16.5s
459:	learn: 105.2445340	total: 3m 10s	remaining: 16.1s
460:	learn: 105.2316382	total: 3m 10s	remaining: 15.7s
461:	learn: 105.1990597	total: 3m 10s	remaining: 15.3s
462:	learn: 104.7964297	total: 3m 11s	remaining: 14.9s
463:	learn: 104.7590363	total: 3m 11s	remaining: 14.5s
464:	learn: 104.6834576	total: 3m 12s	remaining: 14s
465:	learn: 104.6632302	total: 3m 12s	remaining: 13.6s
466:	learn: 104.5948933	total: 3m 12s	remaining: 13.2s
467:	learn: 104.4873380	

103:	learn: 0.2059357	total: 50.8s	remaining: 44s
104:	learn: 0.2041522	total: 51.5s	remaining: 43.6s
105:	learn: 0.2026396	total: 52s	remaining: 43.2s
106:	learn: 0.2010274	total: 52.6s	remaining: 42.8s
107:	learn: 0.1996262	total: 53.2s	remaining: 42.4s
108:	learn: 0.1984051	total: 53.9s	remaining: 42s
109:	learn: 0.1970602	total: 54.5s	remaining: 41.6s
110:	learn: 0.1957778	total: 55s	remaining: 41.1s
111:	learn: 0.1945069	total: 55.5s	remaining: 40.7s
112:	learn: 0.1935556	total: 56s	remaining: 40.1s
113:	learn: 0.1918883	total: 56.5s	remaining: 39.7s
114:	learn: 0.1906593	total: 57s	remaining: 39.2s
115:	learn: 0.1896325	total: 57.5s	remaining: 38.6s
116:	learn: 0.1884391	total: 57.9s	remaining: 38.1s
117:	learn: 0.1865978	total: 58.5s	remaining: 37.7s
118:	learn: 0.1856338	total: 58.9s	remaining: 37.1s
119:	learn: 0.1845908	total: 59.3s	remaining: 36.6s
120:	learn: 0.1834077	total: 59.8s	remaining: 36.1s
121:	learn: 0.1820944	total: 1m	remaining: 35.6s
122:	learn: 0.1808772	total

64:	learn: 0.0623285	total: 30s	remaining: 3m 8s
65:	learn: 0.0612884	total: 30.5s	remaining: 3m 8s
66:	learn: 0.0605580	total: 30.9s	remaining: 3m 7s
67:	learn: 0.0600168	total: 31.4s	remaining: 3m 7s
68:	learn: 0.0595801	total: 31.8s	remaining: 3m 6s
69:	learn: 0.0590696	total: 32.3s	remaining: 3m 6s
70:	learn: 0.0588329	total: 32.7s	remaining: 3m 5s
71:	learn: 0.0582004	total: 33.2s	remaining: 3m 5s
72:	learn: 0.0574514	total: 33.7s	remaining: 3m 4s
73:	learn: 0.0569421	total: 34.1s	remaining: 3m 4s
74:	learn: 0.0564567	total: 34.6s	remaining: 3m 3s
75:	learn: 0.0560465	total: 35s	remaining: 3m 3s
76:	learn: 0.0556626	total: 35.4s	remaining: 3m 2s
77:	learn: 0.0552740	total: 35.9s	remaining: 3m 2s
78:	learn: 0.0550571	total: 36.3s	remaining: 3m 1s
79:	learn: 0.0547668	total: 36.7s	remaining: 3m
80:	learn: 0.0542923	total: 37.2s	remaining: 3m
81:	learn: 0.0537903	total: 37.7s	remaining: 3m
82:	learn: 0.0531140	total: 38.2s	remaining: 2m 59s
83:	learn: 0.0525835	total: 38.6s	remaining

220:	learn: 0.0327708	total: 1m 43s	remaining: 1m 58s
221:	learn: 0.0327389	total: 1m 43s	remaining: 1m 57s
222:	learn: 0.0326705	total: 1m 43s	remaining: 1m 57s
223:	learn: 0.0326199	total: 1m 44s	remaining: 1m 56s
224:	learn: 0.0324896	total: 1m 44s	remaining: 1m 55s
225:	learn: 0.0324178	total: 1m 45s	remaining: 1m 55s
226:	learn: 0.0323658	total: 1m 45s	remaining: 1m 54s
227:	learn: 0.0323219	total: 1m 45s	remaining: 1m 54s
228:	learn: 0.0322786	total: 1m 46s	remaining: 1m 53s
229:	learn: 0.0322178	total: 1m 46s	remaining: 1m 53s
230:	learn: 0.0321441	total: 1m 47s	remaining: 1m 52s
231:	learn: 0.0320800	total: 1m 47s	remaining: 1m 52s
232:	learn: 0.0319730	total: 1m 48s	remaining: 1m 51s
233:	learn: 0.0318617	total: 1m 48s	remaining: 1m 51s
234:	learn: 0.0317658	total: 1m 48s	remaining: 1m 50s
235:	learn: 0.0317266	total: 1m 49s	remaining: 1m 50s
236:	learn: 0.0316031	total: 1m 49s	remaining: 1m 49s
237:	learn: 0.0315566	total: 1m 50s	remaining: 1m 49s
238:	learn: 0.0315088	total:

374:	learn: 0.0257277	total: 2m 48s	remaining: 44.5s
375:	learn: 0.0257118	total: 2m 48s	remaining: 44s
376:	learn: 0.0256908	total: 2m 49s	remaining: 43.5s
377:	learn: 0.0256378	total: 2m 49s	remaining: 43.1s
378:	learn: 0.0256199	total: 2m 50s	remaining: 42.6s
379:	learn: 0.0255870	total: 2m 50s	remaining: 42.2s
380:	learn: 0.0255726	total: 2m 50s	remaining: 41.7s
381:	learn: 0.0255479	total: 2m 51s	remaining: 41.3s
382:	learn: 0.0255265	total: 2m 51s	remaining: 40.8s
383:	learn: 0.0254970	total: 2m 52s	remaining: 40.3s
384:	learn: 0.0254625	total: 2m 52s	remaining: 39.9s
385:	learn: 0.0254316	total: 2m 53s	remaining: 39.5s
386:	learn: 0.0254027	total: 2m 53s	remaining: 39s
387:	learn: 0.0253870	total: 2m 53s	remaining: 38.5s
388:	learn: 0.0253575	total: 2m 54s	remaining: 38.1s
389:	learn: 0.0253337	total: 2m 54s	remaining: 37.6s
390:	learn: 0.0253223	total: 2m 55s	remaining: 37.2s
391:	learn: 0.0252975	total: 2m 55s	remaining: 36.7s
392:	learn: 0.0252646	total: 2m 56s	remaining: 36.

54:	learn: 0.5629321	total: 50.9s	remaining: 3m 25s
55:	learn: 0.5555387	total: 51.7s	remaining: 3m 24s
56:	learn: 0.5476601	total: 52.6s	remaining: 3m 23s
57:	learn: 0.5390800	total: 53.4s	remaining: 3m 21s
58:	learn: 0.5317454	total: 54.3s	remaining: 3m 20s
59:	learn: 0.5251906	total: 55.1s	remaining: 3m 19s
60:	learn: 0.5173902	total: 55.9s	remaining: 3m 18s
61:	learn: 0.5102490	total: 56.8s	remaining: 3m 16s
62:	learn: 0.5034636	total: 57.6s	remaining: 3m 15s
63:	learn: 0.4969358	total: 58.5s	remaining: 3m 14s
64:	learn: 0.4908770	total: 59.3s	remaining: 3m 13s
65:	learn: 0.4854662	total: 1m	remaining: 3m 12s
66:	learn: 0.4790765	total: 1m	remaining: 3m 10s
67:	learn: 0.4726027	total: 1m 1s	remaining: 3m 9s
68:	learn: 0.4671548	total: 1m 2s	remaining: 3m 8s
69:	learn: 0.4610372	total: 1m 3s	remaining: 3m 7s
70:	learn: 0.4548697	total: 1m 4s	remaining: 3m 6s
71:	learn: 0.4499377	total: 1m 4s	remaining: 3m 4s
72:	learn: 0.4457768	total: 1m 5s	remaining: 3m 3s
73:	learn: 0.4404076	tot

209:	learn: 0.1872889	total: 2m 56s	remaining: 56.4s
210:	learn: 0.1865101	total: 2m 57s	remaining: 55.5s
211:	learn: 0.1858871	total: 2m 58s	remaining: 54.7s
212:	learn: 0.1850606	total: 2m 59s	remaining: 53.8s
213:	learn: 0.1845261	total: 2m 59s	remaining: 53s
214:	learn: 0.1839287	total: 3m	remaining: 52.1s
215:	learn: 0.1834681	total: 3m 1s	remaining: 51.3s
216:	learn: 0.1828347	total: 3m 2s	remaining: 50.4s
217:	learn: 0.1821636	total: 3m 3s	remaining: 49.6s
218:	learn: 0.1813412	total: 3m 3s	remaining: 48.7s
219:	learn: 0.1805691	total: 3m 4s	remaining: 47.9s
220:	learn: 0.1799305	total: 3m 5s	remaining: 47s
221:	learn: 0.1794753	total: 3m 6s	remaining: 46.2s
222:	learn: 0.1786217	total: 3m 7s	remaining: 45.3s
223:	learn: 0.1779792	total: 3m 7s	remaining: 44.5s
224:	learn: 0.1775660	total: 3m 8s	remaining: 43.6s
225:	learn: 0.1770075	total: 3m 9s	remaining: 42.8s
226:	learn: 0.1763911	total: 3m 10s	remaining: 41.9s
227:	learn: 0.1759886	total: 3m 11s	remaining: 41.1s
228:	learn: 

85:	learn: 0.1097670	total: 3m 25s	remaining: 7m 49s
86:	learn: 0.1087992	total: 3m 27s	remaining: 7m 47s
87:	learn: 0.1074677	total: 3m 29s	remaining: 7m 44s
88:	learn: 0.1063709	total: 3m 32s	remaining: 7m 42s
89:	learn: 0.1049029	total: 3m 34s	remaining: 7m 40s
90:	learn: 0.1039039	total: 3m 36s	remaining: 7m 37s
91:	learn: 0.1028606	total: 3m 39s	remaining: 7m 35s
92:	learn: 0.1018312	total: 3m 41s	remaining: 7m 32s
93:	learn: 0.1010206	total: 3m 44s	remaining: 7m 30s
94:	learn: 0.1001227	total: 3m 46s	remaining: 7m 27s
95:	learn: 0.0990940	total: 3m 48s	remaining: 7m 26s
96:	learn: 0.0983506	total: 3m 51s	remaining: 7m 23s
97:	learn: 0.0974612	total: 3m 53s	remaining: 7m 21s
98:	learn: 0.0965348	total: 3m 56s	remaining: 7m 18s
99:	learn: 0.0954742	total: 3m 58s	remaining: 7m 16s
100:	learn: 0.0946222	total: 4m	remaining: 7m 14s
101:	learn: 0.0940043	total: 4m 3s	remaining: 7m 11s
102:	learn: 0.0933569	total: 4m 5s	remaining: 7m 9s
103:	learn: 0.0923884	total: 4m 7s	remaining: 7m 6

239:	learn: 0.0481943	total: 9m 32s	remaining: 1m 42s
240:	learn: 0.0480360	total: 9m 34s	remaining: 1m 40s
241:	learn: 0.0479117	total: 9m 37s	remaining: 1m 37s
242:	learn: 0.0478103	total: 9m 39s	remaining: 1m 35s
243:	learn: 0.0476532	total: 9m 41s	remaining: 1m 33s
244:	learn: 0.0475223	total: 9m 44s	remaining: 1m 30s
245:	learn: 0.0474158	total: 9m 46s	remaining: 1m 28s
246:	learn: 0.0472815	total: 9m 49s	remaining: 1m 25s
247:	learn: 0.0471869	total: 9m 51s	remaining: 1m 23s
248:	learn: 0.0470821	total: 9m 53s	remaining: 1m 21s
249:	learn: 0.0469079	total: 9m 56s	remaining: 1m 18s
250:	learn: 0.0468157	total: 9m 58s	remaining: 1m 16s
251:	learn: 0.0466615	total: 10m 1s	remaining: 1m 13s
252:	learn: 0.0465377	total: 10m 3s	remaining: 1m 11s
253:	learn: 0.0464496	total: 10m 5s	remaining: 1m 9s
254:	learn: 0.0463490	total: 10m 8s	remaining: 1m 6s
255:	learn: 0.0462418	total: 10m 10s	remaining: 1m 4s
256:	learn: 0.0461403	total: 10m 13s	remaining: 1m 2s
257:	learn: 0.0460229	total: 1

110:	learn: 0.0729418	total: 40.5s	remaining: 1m 40s
111:	learn: 0.0724687	total: 40.8s	remaining: 1m 39s
112:	learn: 0.0721538	total: 41.2s	remaining: 1m 39s
113:	learn: 0.0719497	total: 41.5s	remaining: 1m 39s
114:	learn: 0.0716479	total: 41.8s	remaining: 1m 38s
115:	learn: 0.0709841	total: 42.3s	remaining: 1m 38s
116:	learn: 0.0704906	total: 42.7s	remaining: 1m 38s
117:	learn: 0.0703840	total: 43s	remaining: 1m 37s
118:	learn: 0.0700340	total: 43.3s	remaining: 1m 37s
119:	learn: 0.0694505	total: 43.7s	remaining: 1m 36s
120:	learn: 0.0689455	total: 44s	remaining: 1m 36s
121:	learn: 0.0684505	total: 44.4s	remaining: 1m 36s
122:	learn: 0.0681780	total: 44.8s	remaining: 1m 35s
123:	learn: 0.0678399	total: 45.2s	remaining: 1m 35s
124:	learn: 0.0675988	total: 45.5s	remaining: 1m 35s
125:	learn: 0.0672980	total: 45.9s	remaining: 1m 34s
126:	learn: 0.0668102	total: 46.3s	remaining: 1m 34s
127:	learn: 0.0665594	total: 46.6s	remaining: 1m 34s
128:	learn: 0.0661855	total: 47s	remaining: 1m 33s

266:	learn: 0.0454204	total: 1m 35s	remaining: 42.6s
267:	learn: 0.0453966	total: 1m 35s	remaining: 42.3s
268:	learn: 0.0453166	total: 1m 36s	remaining: 41.9s
269:	learn: 0.0452784	total: 1m 36s	remaining: 41.5s
270:	learn: 0.0450658	total: 1m 37s	remaining: 41.2s
271:	learn: 0.0450190	total: 1m 37s	remaining: 40.8s
272:	learn: 0.0449522	total: 1m 37s	remaining: 40.5s
273:	learn: 0.0449161	total: 1m 38s	remaining: 40.1s
274:	learn: 0.0447771	total: 1m 38s	remaining: 39.7s
275:	learn: 0.0446905	total: 1m 38s	remaining: 39.4s
276:	learn: 0.0445783	total: 1m 39s	remaining: 39s
277:	learn: 0.0445380	total: 1m 39s	remaining: 38.7s
278:	learn: 0.0445251	total: 1m 39s	remaining: 38.3s
279:	learn: 0.0444528	total: 1m 40s	remaining: 37.9s
280:	learn: 0.0443240	total: 1m 40s	remaining: 37.6s
281:	learn: 0.0442540	total: 1m 40s	remaining: 37.2s
282:	learn: 0.0442321	total: 1m 41s	remaining: 36.9s
283:	learn: 0.0441999	total: 1m 41s	remaining: 36.5s
284:	learn: 0.0441655	total: 1m 41s	remaining: 3

33:	learn: 0.0585288	total: 53.1s	remaining: 10m 44s
34:	learn: 0.0573441	total: 54.7s	remaining: 10m 42s
35:	learn: 0.0560685	total: 56.3s	remaining: 10m 40s
36:	learn: 0.0545446	total: 57.8s	remaining: 10m 38s
37:	learn: 0.0538234	total: 59.3s	remaining: 10m 36s
38:	learn: 0.0523152	total: 1m	remaining: 10m 35s
39:	learn: 0.0510771	total: 1m 2s	remaining: 10m 34s
40:	learn: 0.0499832	total: 1m 4s	remaining: 10m 33s
41:	learn: 0.0490173	total: 1m 5s	remaining: 10m 31s
42:	learn: 0.0481770	total: 1m 7s	remaining: 10m 30s
43:	learn: 0.0477107	total: 1m 8s	remaining: 10m 28s
44:	learn: 0.0466051	total: 1m 10s	remaining: 10m 26s
45:	learn: 0.0458866	total: 1m 11s	remaining: 10m 25s
46:	learn: 0.0448292	total: 1m 13s	remaining: 10m 24s
47:	learn: 0.0442792	total: 1m 15s	remaining: 10m 23s
48:	learn: 0.0434660	total: 1m 16s	remaining: 10m 22s
49:	learn: 0.0427156	total: 1m 18s	remaining: 10m 20s
50:	learn: 0.0423827	total: 1m 19s	remaining: 10m 18s
51:	learn: 0.0417867	total: 1m 21s	remaini

187:	learn: 0.0166735	total: 4m 59s	remaining: 6m 50s
188:	learn: 0.0166129	total: 5m	remaining: 6m 49s
189:	learn: 0.0165446	total: 5m 2s	remaining: 6m 47s
190:	learn: 0.0164527	total: 5m 4s	remaining: 6m 45s
191:	learn: 0.0163937	total: 5m 5s	remaining: 6m 44s
192:	learn: 0.0163535	total: 5m 7s	remaining: 6m 42s
193:	learn: 0.0162649	total: 5m 8s	remaining: 6m 41s
194:	learn: 0.0162191	total: 5m 10s	remaining: 6m 39s
195:	learn: 0.0161853	total: 5m 12s	remaining: 6m 38s
196:	learn: 0.0161321	total: 5m 13s	remaining: 6m 36s
197:	learn: 0.0161165	total: 5m 14s	remaining: 6m 33s
198:	learn: 0.0160653	total: 5m 16s	remaining: 6m 32s
199:	learn: 0.0159919	total: 5m 17s	remaining: 6m 30s
200:	learn: 0.0159057	total: 5m 19s	remaining: 6m 29s
201:	learn: 0.0158401	total: 5m 20s	remaining: 6m 27s
202:	learn: 0.0157808	total: 5m 22s	remaining: 6m 26s
203:	learn: 0.0157517	total: 5m 24s	remaining: 6m 24s
204:	learn: 0.0156543	total: 5m 25s	remaining: 6m 23s
205:	learn: 0.0156243	total: 5m 27s	r

340:	learn: 0.0100165	total: 8m 59s	remaining: 2m 46s
341:	learn: 0.0099876	total: 9m 1s	remaining: 2m 44s
342:	learn: 0.0099650	total: 9m 2s	remaining: 2m 43s
343:	learn: 0.0099561	total: 9m 4s	remaining: 2m 41s
344:	learn: 0.0099316	total: 9m 5s	remaining: 2m 39s
345:	learn: 0.0098897	total: 9m 7s	remaining: 2m 38s
346:	learn: 0.0098471	total: 9m 9s	remaining: 2m 36s
347:	learn: 0.0098184	total: 9m 10s	remaining: 2m 35s
348:	learn: 0.0098097	total: 9m 12s	remaining: 2m 33s
349:	learn: 0.0097904	total: 9m 13s	remaining: 2m 31s
350:	learn: 0.0097644	total: 9m 15s	remaining: 2m 30s
351:	learn: 0.0097466	total: 9m 16s	remaining: 2m 28s
352:	learn: 0.0097274	total: 9m 18s	remaining: 2m 27s
353:	learn: 0.0097110	total: 9m 19s	remaining: 2m 25s
354:	learn: 0.0096861	total: 9m 21s	remaining: 2m 23s
355:	learn: 0.0096573	total: 9m 23s	remaining: 2m 22s
356:	learn: 0.0096239	total: 9m 24s	remaining: 2m 20s
357:	learn: 0.0095838	total: 9m 26s	remaining: 2m 19s
358:	learn: 0.0095616	total: 9m 27

45:	learn: 0.1029009	total: 44.7s	remaining: 1m 15s
46:	learn: 0.1017582	total: 45.6s	remaining: 1m 14s
47:	learn: 0.0995221	total: 46.5s	remaining: 1m 13s
48:	learn: 0.0980911	total: 47.6s	remaining: 1m 12s
49:	learn: 0.0960240	total: 48.6s	remaining: 1m 11s
50:	learn: 0.0940401	total: 49.6s	remaining: 1m 11s
51:	learn: 0.0929134	total: 50.6s	remaining: 1m 10s
52:	learn: 0.0917632	total: 51.6s	remaining: 1m 9s
53:	learn: 0.0903905	total: 52.5s	remaining: 1m 8s
54:	learn: 0.0891650	total: 53.5s	remaining: 1m 7s
55:	learn: 0.0880269	total: 54.5s	remaining: 1m 6s
56:	learn: 0.0868229	total: 55.4s	remaining: 1m 5s
57:	learn: 0.0860234	total: 56.4s	remaining: 1m 4s
58:	learn: 0.0846309	total: 57.4s	remaining: 1m 3s
59:	learn: 0.0834063	total: 58.3s	remaining: 1m 2s
60:	learn: 0.0824401	total: 59.3s	remaining: 1m 1s
61:	learn: 0.0810683	total: 1m	remaining: 1m
62:	learn: 0.0801171	total: 1m 1s	remaining: 59.2s
63:	learn: 0.0791265	total: 1m 2s	remaining: 58.2s
64:	learn: 0.0779288	total: 1m

76:	learn: 0.5823264	total: 32.9s	remaining: 1m 20s
77:	learn: 0.5773305	total: 33.4s	remaining: 1m 19s
78:	learn: 0.5711450	total: 33.8s	remaining: 1m 19s
79:	learn: 0.5650970	total: 34.2s	remaining: 1m 19s
80:	learn: 0.5597481	total: 34.6s	remaining: 1m 18s
81:	learn: 0.5539905	total: 35.1s	remaining: 1m 18s
82:	learn: 0.5496095	total: 35.5s	remaining: 1m 17s
83:	learn: 0.5449921	total: 36s	remaining: 1m 17s
84:	learn: 0.5396433	total: 36.4s	remaining: 1m 17s
85:	learn: 0.5359467	total: 36.9s	remaining: 1m 16s
86:	learn: 0.5312063	total: 37.3s	remaining: 1m 16s
87:	learn: 0.5264357	total: 37.8s	remaining: 1m 15s
88:	learn: 0.5221546	total: 38.2s	remaining: 1m 15s
89:	learn: 0.5185144	total: 38.7s	remaining: 1m 15s
90:	learn: 0.5147889	total: 39.1s	remaining: 1m 14s
91:	learn: 0.5109749	total: 39.6s	remaining: 1m 14s
92:	learn: 0.5065050	total: 40s	remaining: 1m 14s
93:	learn: 0.5023023	total: 40.5s	remaining: 1m 13s
94:	learn: 0.4989814	total: 40.9s	remaining: 1m 13s
95:	learn: 0.495

234:	learn: 0.2720406	total: 1m 39s	remaining: 12.7s
235:	learn: 0.2712633	total: 1m 40s	remaining: 12.3s
236:	learn: 0.2702657	total: 1m 40s	remaining: 11.9s
237:	learn: 0.2695021	total: 1m 40s	remaining: 11.4s
238:	learn: 0.2685811	total: 1m 41s	remaining: 11s
239:	learn: 0.2678816	total: 1m 41s	remaining: 10.6s
240:	learn: 0.2672592	total: 1m 42s	remaining: 10.2s
241:	learn: 0.2664735	total: 1m 42s	remaining: 9.75s
242:	learn: 0.2658415	total: 1m 42s	remaining: 9.32s
243:	learn: 0.2650440	total: 1m 43s	remaining: 8.9s
244:	learn: 0.2641514	total: 1m 43s	remaining: 8.48s
245:	learn: 0.2632679	total: 1m 44s	remaining: 8.06s
246:	learn: 0.2624714	total: 1m 44s	remaining: 7.63s
247:	learn: 0.2618857	total: 1m 45s	remaining: 7.21s
248:	learn: 0.2613080	total: 1m 45s	remaining: 6.78s
249:	learn: 0.2605156	total: 1m 45s	remaining: 6.36s
250:	learn: 0.2596106	total: 1m 46s	remaining: 5.93s
251:	learn: 0.2587298	total: 1m 46s	remaining: 5.51s
252:	learn: 0.2580258	total: 1m 47s	remaining: 5.

124:	learn: 0.3736951	total: 41.5s	remaining: 1m 35s
125:	learn: 0.3722751	total: 41.8s	remaining: 1m 35s
126:	learn: 0.3708099	total: 42.1s	remaining: 1m 34s
127:	learn: 0.3693221	total: 42.5s	remaining: 1m 34s
128:	learn: 0.3666196	total: 42.8s	remaining: 1m 34s
129:	learn: 0.3652390	total: 43.1s	remaining: 1m 33s
130:	learn: 0.3643696	total: 43.5s	remaining: 1m 33s
131:	learn: 0.3632940	total: 43.8s	remaining: 1m 33s
132:	learn: 0.3617033	total: 44.1s	remaining: 1m 32s
133:	learn: 0.3599559	total: 44.4s	remaining: 1m 32s
134:	learn: 0.3573266	total: 44.8s	remaining: 1m 32s
135:	learn: 0.3560012	total: 45.1s	remaining: 1m 31s
136:	learn: 0.3545227	total: 45.4s	remaining: 1m 31s
137:	learn: 0.3526696	total: 45.7s	remaining: 1m 31s
138:	learn: 0.3517244	total: 46s	remaining: 1m 30s
139:	learn: 0.3509561	total: 46.4s	remaining: 1m 30s
140:	learn: 0.3491951	total: 46.7s	remaining: 1m 30s
141:	learn: 0.3479891	total: 47s	remaining: 1m 29s
142:	learn: 0.3463948	total: 47.4s	remaining: 1m 2

280:	learn: 0.2209196	total: 1m 33s	remaining: 43.7s
281:	learn: 0.2202139	total: 1m 33s	remaining: 43.4s
282:	learn: 0.2199342	total: 1m 33s	remaining: 43.1s
283:	learn: 0.2193782	total: 1m 34s	remaining: 42.7s
284:	learn: 0.2191398	total: 1m 34s	remaining: 42.4s
285:	learn: 0.2183941	total: 1m 34s	remaining: 42.1s
286:	learn: 0.2179827	total: 1m 35s	remaining: 41.7s
287:	learn: 0.2177040	total: 1m 35s	remaining: 41.4s
288:	learn: 0.2175042	total: 1m 35s	remaining: 41s
289:	learn: 0.2169805	total: 1m 36s	remaining: 40.7s
290:	learn: 0.2167125	total: 1m 36s	remaining: 40.4s
291:	learn: 0.2163266	total: 1m 36s	remaining: 40s
292:	learn: 0.2158923	total: 1m 36s	remaining: 39.7s
293:	learn: 0.2155524	total: 1m 37s	remaining: 39.4s
294:	learn: 0.2149721	total: 1m 37s	remaining: 39s
295:	learn: 0.2144744	total: 1m 37s	remaining: 38.7s
296:	learn: 0.2142119	total: 1m 38s	remaining: 38.4s
297:	learn: 0.2138822	total: 1m 38s	remaining: 38s
298:	learn: 0.2129180	total: 1m 38s	remaining: 37.7s
2

20:	learn: 1.5436958	total: 52.3s	remaining: 3m 16s
21:	learn: 1.5190662	total: 54.8s	remaining: 3m 14s
22:	learn: 1.4947096	total: 57.3s	remaining: 3m 11s
23:	learn: 1.4712520	total: 59.9s	remaining: 3m 9s
24:	learn: 1.4489137	total: 1m 2s	remaining: 3m 7s
25:	learn: 1.4274381	total: 1m 4s	remaining: 3m 4s
26:	learn: 1.4064198	total: 1m 7s	remaining: 3m 1s
27:	learn: 1.3856890	total: 1m 9s	remaining: 2m 59s
28:	learn: 1.3662877	total: 1m 12s	remaining: 2m 56s
29:	learn: 1.3475833	total: 1m 14s	remaining: 2m 53s
30:	learn: 1.3288709	total: 1m 16s	remaining: 2m 51s
31:	learn: 1.3104587	total: 1m 19s	remaining: 2m 48s
32:	learn: 1.2926153	total: 1m 21s	remaining: 2m 46s
33:	learn: 1.2753675	total: 1m 24s	remaining: 2m 43s
34:	learn: 1.2588063	total: 1m 26s	remaining: 2m 41s
35:	learn: 1.2427911	total: 1m 29s	remaining: 2m 38s
36:	learn: 1.2265205	total: 1m 31s	remaining: 2m 36s
37:	learn: 1.2115149	total: 1m 34s	remaining: 2m 33s
38:	learn: 1.1965010	total: 1m 36s	remaining: 2m 30s
39:	l

72:	learn: 0.0635348	total: 2m 55s	remaining: 17m 5s
73:	learn: 0.0629992	total: 2m 57s	remaining: 17m 3s
74:	learn: 0.0623195	total: 3m	remaining: 17m 1s
75:	learn: 0.0614944	total: 3m 2s	remaining: 16m 59s
76:	learn: 0.0609820	total: 3m 5s	remaining: 16m 56s
77:	learn: 0.0601752	total: 3m 7s	remaining: 16m 54s
78:	learn: 0.0597206	total: 3m 9s	remaining: 16m 51s
79:	learn: 0.0592751	total: 3m 12s	remaining: 16m 49s
80:	learn: 0.0587980	total: 3m 14s	remaining: 16m 46s
81:	learn: 0.0581929	total: 3m 17s	remaining: 16m 44s
82:	learn: 0.0577621	total: 3m 19s	remaining: 16m 41s
83:	learn: 0.0572547	total: 3m 21s	remaining: 16m 39s
84:	learn: 0.0566568	total: 3m 24s	remaining: 16m 36s
85:	learn: 0.0563611	total: 3m 26s	remaining: 16m 33s
86:	learn: 0.0557319	total: 3m 28s	remaining: 16m 31s
87:	learn: 0.0550678	total: 3m 31s	remaining: 16m 28s
88:	learn: 0.0545522	total: 3m 33s	remaining: 16m 26s
89:	learn: 0.0541624	total: 3m 36s	remaining: 16m 24s
90:	learn: 0.0536191	total: 3m 38s	rema

223:	learn: 0.0288310	total: 8m 55s	remaining: 11m
224:	learn: 0.0287588	total: 8m 58s	remaining: 10m 57s
225:	learn: 0.0286940	total: 9m	remaining: 10m 55s
226:	learn: 0.0286017	total: 9m 3s	remaining: 10m 53s
227:	learn: 0.0285603	total: 9m 5s	remaining: 10m 50s
228:	learn: 0.0285200	total: 9m 7s	remaining: 10m 48s
229:	learn: 0.0284184	total: 9m 10s	remaining: 10m 45s
230:	learn: 0.0283131	total: 9m 12s	remaining: 10m 43s
231:	learn: 0.0282552	total: 9m 14s	remaining: 10m 41s
232:	learn: 0.0282054	total: 9m 17s	remaining: 10m 38s
233:	learn: 0.0281549	total: 9m 19s	remaining: 10m 36s
234:	learn: 0.0280817	total: 9m 22s	remaining: 10m 33s
235:	learn: 0.0279725	total: 9m 24s	remaining: 10m 31s
236:	learn: 0.0278882	total: 9m 26s	remaining: 10m 29s
237:	learn: 0.0278388	total: 9m 29s	remaining: 10m 26s
238:	learn: 0.0277625	total: 9m 31s	remaining: 10m 24s
239:	learn: 0.0276644	total: 9m 34s	remaining: 10m 22s
240:	learn: 0.0275878	total: 9m 36s	remaining: 10m 19s
241:	learn: 0.0275351

374:	learn: 0.0207190	total: 14m 58s	remaining: 4m 59s
375:	learn: 0.0206663	total: 15m 1s	remaining: 4m 57s
376:	learn: 0.0206433	total: 15m 3s	remaining: 4m 54s
377:	learn: 0.0206117	total: 15m 5s	remaining: 4m 52s
378:	learn: 0.0205834	total: 15m 8s	remaining: 4m 49s
379:	learn: 0.0205234	total: 15m 10s	remaining: 4m 47s
380:	learn: 0.0204866	total: 15m 12s	remaining: 4m 45s
381:	learn: 0.0204527	total: 15m 15s	remaining: 4m 42s
382:	learn: 0.0203943	total: 15m 17s	remaining: 4m 40s
383:	learn: 0.0203692	total: 15m 20s	remaining: 4m 37s
384:	learn: 0.0203416	total: 15m 22s	remaining: 4m 35s
385:	learn: 0.0203184	total: 15m 24s	remaining: 4m 33s
386:	learn: 0.0202835	total: 15m 27s	remaining: 4m 30s
387:	learn: 0.0202423	total: 15m 29s	remaining: 4m 28s
388:	learn: 0.0202036	total: 15m 32s	remaining: 4m 25s
389:	learn: 0.0201819	total: 15m 34s	remaining: 4m 23s
390:	learn: 0.0201391	total: 15m 36s	remaining: 4m 21s
391:	learn: 0.0200962	total: 15m 39s	remaining: 4m 18s
392:	learn: 0.

23:	learn: 0.1827027	total: 57.6s	remaining: 3m 2s
24:	learn: 0.1751231	total: 60s	remaining: 2m 59s
25:	learn: 0.1690226	total: 1m 2s	remaining: 2m 57s
26:	learn: 0.1619651	total: 1m 4s	remaining: 2m 54s
27:	learn: 0.1570063	total: 1m 7s	remaining: 2m 52s
28:	learn: 0.1504284	total: 1m 9s	remaining: 2m 49s
29:	learn: 0.1453151	total: 1m 11s	remaining: 2m 47s
30:	learn: 0.1406133	total: 1m 14s	remaining: 2m 45s
31:	learn: 0.1364661	total: 1m 16s	remaining: 2m 42s
32:	learn: 0.1324738	total: 1m 18s	remaining: 2m 39s
33:	learn: 0.1284730	total: 1m 21s	remaining: 2m 37s
34:	learn: 0.1249433	total: 1m 23s	remaining: 2m 34s
35:	learn: 0.1214121	total: 1m 25s	remaining: 2m 32s
36:	learn: 0.1186320	total: 1m 28s	remaining: 2m 30s
37:	learn: 0.1155732	total: 1m 30s	remaining: 2m 27s
38:	learn: 0.1128769	total: 1m 32s	remaining: 2m 25s
39:	learn: 0.1102029	total: 1m 35s	remaining: 2m 22s
40:	learn: 0.1079996	total: 1m 37s	remaining: 2m 20s
41:	learn: 0.1058427	total: 1m 40s	remaining: 2m 18s
42

73:	learn: 9748.6412819	total: 29.7s	remaining: 26.1s
74:	learn: 9671.7071703	total: 30.1s	remaining: 25.6s
75:	learn: 9561.1398090	total: 30.4s	remaining: 25.2s
76:	learn: 9472.3962908	total: 30.8s	remaining: 24.8s
77:	learn: 9401.1364344	total: 31.2s	remaining: 24.4s
78:	learn: 9365.9157359	total: 31.5s	remaining: 23.9s
79:	learn: 9307.8644123	total: 31.9s	remaining: 23.5s
80:	learn: 9297.8889190	total: 32.3s	remaining: 23.1s
81:	learn: 9237.9654730	total: 32.7s	remaining: 22.7s
82:	learn: 9222.3328123	total: 33.1s	remaining: 22.3s
83:	learn: 8917.1781844	total: 33.4s	remaining: 21.9s
84:	learn: 8652.3528893	total: 33.9s	remaining: 21.5s
85:	learn: 8618.2855708	total: 34.2s	remaining: 21.1s
86:	learn: 8596.4997810	total: 34.6s	remaining: 20.7s
87:	learn: 8455.8037525	total: 35s	remaining: 20.3s
88:	learn: 8312.8977388	total: 35.3s	remaining: 19.8s
89:	learn: 8236.3818089	total: 35.7s	remaining: 19.4s
90:	learn: 8227.1550492	total: 36.1s	remaining: 19s
91:	learn: 8075.1058215	total: 3

85:	learn: 0.0512524	total: 1m 42s	remaining: 8m 14s
86:	learn: 0.0510084	total: 1m 43s	remaining: 8m 13s
87:	learn: 0.0506173	total: 1m 45s	remaining: 8m 11s
88:	learn: 0.0500012	total: 1m 46s	remaining: 8m 10s
89:	learn: 0.0496427	total: 1m 47s	remaining: 8m 9s
90:	learn: 0.0494271	total: 1m 48s	remaining: 8m 7s
91:	learn: 0.0489363	total: 1m 49s	remaining: 8m 6s
92:	learn: 0.0486774	total: 1m 50s	remaining: 8m 5s
93:	learn: 0.0484960	total: 1m 52s	remaining: 8m 4s
94:	learn: 0.0480540	total: 1m 53s	remaining: 8m 3s
95:	learn: 0.0479466	total: 1m 54s	remaining: 8m 1s
96:	learn: 0.0475331	total: 1m 55s	remaining: 8m
97:	learn: 0.0470056	total: 1m 56s	remaining: 7m 59s
98:	learn: 0.0466886	total: 1m 57s	remaining: 7m 57s
99:	learn: 0.0463470	total: 1m 59s	remaining: 7m 56s
100:	learn: 0.0460625	total: 2m	remaining: 7m 55s
101:	learn: 0.0457720	total: 2m 1s	remaining: 7m 54s
102:	learn: 0.0453876	total: 2m 2s	remaining: 7m 52s
103:	learn: 0.0451806	total: 2m 3s	remaining: 7m 51s
104:	le

239:	learn: 0.0281625	total: 4m 44s	remaining: 5m 8s
240:	learn: 0.0281050	total: 4m 45s	remaining: 5m 7s
241:	learn: 0.0280513	total: 4m 47s	remaining: 5m 5s
242:	learn: 0.0279739	total: 4m 48s	remaining: 5m 4s
243:	learn: 0.0279081	total: 4m 49s	remaining: 5m 3s
244:	learn: 0.0278714	total: 4m 50s	remaining: 5m 2s
245:	learn: 0.0277994	total: 4m 51s	remaining: 5m 1s
246:	learn: 0.0277679	total: 4m 52s	remaining: 4m 59s
247:	learn: 0.0277260	total: 4m 54s	remaining: 4m 58s
248:	learn: 0.0276770	total: 4m 55s	remaining: 4m 57s
249:	learn: 0.0276326	total: 4m 56s	remaining: 4m 56s
250:	learn: 0.0275718	total: 4m 57s	remaining: 4m 55s
251:	learn: 0.0275534	total: 4m 58s	remaining: 4m 53s
252:	learn: 0.0274906	total: 4m 59s	remaining: 4m 52s
253:	learn: 0.0274353	total: 5m 1s	remaining: 4m 51s
254:	learn: 0.0273397	total: 5m 2s	remaining: 4m 50s
255:	learn: 0.0272821	total: 5m 3s	remaining: 4m 49s
256:	learn: 0.0272304	total: 5m 4s	remaining: 4m 48s
257:	learn: 0.0271916	total: 5m 5s	rema

392:	learn: 0.0214008	total: 7m 48s	remaining: 2m 7s
393:	learn: 0.0213681	total: 7m 49s	remaining: 2m 6s
394:	learn: 0.0213312	total: 7m 50s	remaining: 2m 5s
395:	learn: 0.0212968	total: 7m 51s	remaining: 2m 3s
396:	learn: 0.0212676	total: 7m 53s	remaining: 2m 2s
397:	learn: 0.0212329	total: 7m 54s	remaining: 2m 1s
398:	learn: 0.0212180	total: 7m 55s	remaining: 2m
399:	learn: 0.0211787	total: 7m 56s	remaining: 1m 59s
400:	learn: 0.0211464	total: 7m 57s	remaining: 1m 57s
401:	learn: 0.0211256	total: 7m 59s	remaining: 1m 56s
402:	learn: 0.0211014	total: 8m	remaining: 1m 55s
403:	learn: 0.0210712	total: 8m 1s	remaining: 1m 54s
404:	learn: 0.0210517	total: 8m 2s	remaining: 1m 53s
405:	learn: 0.0210310	total: 8m 3s	remaining: 1m 52s
406:	learn: 0.0209994	total: 8m 5s	remaining: 1m 50s
407:	learn: 0.0209515	total: 8m 6s	remaining: 1m 49s
408:	learn: 0.0209130	total: 8m 7s	remaining: 1m 48s
409:	learn: 0.0208717	total: 8m 8s	remaining: 1m 47s
410:	learn: 0.0208385	total: 8m 10s	remaining: 1m

43:	learn: 0.3512094	total: 1m 46s	remaining: 18m 28s
44:	learn: 0.3428483	total: 1m 49s	remaining: 18m 26s
45:	learn: 0.3353308	total: 1m 51s	remaining: 18m 23s
46:	learn: 0.3278988	total: 1m 54s	remaining: 18m 20s
47:	learn: 0.3213988	total: 1m 56s	remaining: 18m 18s
48:	learn: 0.3144272	total: 1m 59s	remaining: 18m 15s
49:	learn: 0.3070441	total: 2m 1s	remaining: 18m 12s
50:	learn: 0.3006929	total: 2m 3s	remaining: 18m 9s
51:	learn: 0.2942642	total: 2m 6s	remaining: 18m 6s
52:	learn: 0.2884049	total: 2m 8s	remaining: 18m 3s
53:	learn: 0.2828163	total: 2m 10s	remaining: 18m 1s
54:	learn: 0.2775049	total: 2m 13s	remaining: 17m 57s
55:	learn: 0.2719343	total: 2m 15s	remaining: 17m 55s
56:	learn: 0.2671043	total: 2m 17s	remaining: 17m 52s
57:	learn: 0.2624315	total: 2m 20s	remaining: 17m 49s
58:	learn: 0.2574547	total: 2m 22s	remaining: 17m 46s
59:	learn: 0.2530606	total: 2m 25s	remaining: 17m 43s
60:	learn: 0.2484266	total: 2m 27s	remaining: 17m 40s
61:	learn: 0.2441129	total: 2m 29s	r

195:	learn: 0.0835719	total: 7m 51s	remaining: 12m 10s
196:	learn: 0.0831889	total: 7m 53s	remaining: 12m 8s
197:	learn: 0.0827910	total: 7m 56s	remaining: 12m 6s
198:	learn: 0.0823464	total: 7m 58s	remaining: 12m 4s
199:	learn: 0.0818757	total: 8m 1s	remaining: 12m 1s
200:	learn: 0.0815460	total: 8m 3s	remaining: 11m 59s
201:	learn: 0.0812272	total: 8m 5s	remaining: 11m 56s
202:	learn: 0.0808852	total: 8m 8s	remaining: 11m 54s
203:	learn: 0.0805247	total: 8m 10s	remaining: 11m 52s
204:	learn: 0.0801878	total: 8m 13s	remaining: 11m 49s
205:	learn: 0.0797898	total: 8m 15s	remaining: 11m 47s
206:	learn: 0.0795095	total: 8m 17s	remaining: 11m 44s
207:	learn: 0.0791996	total: 8m 20s	remaining: 11m 42s
208:	learn: 0.0788140	total: 8m 22s	remaining: 11m 39s
209:	learn: 0.0784457	total: 8m 25s	remaining: 11m 37s
210:	learn: 0.0782009	total: 8m 27s	remaining: 11m 35s
211:	learn: 0.0779474	total: 8m 29s	remaining: 11m 32s
212:	learn: 0.0776995	total: 8m 32s	remaining: 11m 30s
213:	learn: 0.0774

346:	learn: 0.0534873	total: 13m 54s	remaining: 6m 7s
347:	learn: 0.0533865	total: 13m 56s	remaining: 6m 5s
348:	learn: 0.0532655	total: 13m 59s	remaining: 6m 3s
349:	learn: 0.0531783	total: 14m 1s	remaining: 6m
350:	learn: 0.0530993	total: 14m 4s	remaining: 5m 58s
351:	learn: 0.0530117	total: 14m 6s	remaining: 5m 55s
352:	learn: 0.0528617	total: 14m 9s	remaining: 5m 53s
353:	learn: 0.0527544	total: 14m 11s	remaining: 5m 51s
354:	learn: 0.0526299	total: 14m 13s	remaining: 5m 48s
355:	learn: 0.0525197	total: 14m 16s	remaining: 5m 46s
356:	learn: 0.0523828	total: 14m 18s	remaining: 5m 44s
357:	learn: 0.0522649	total: 14m 21s	remaining: 5m 41s
358:	learn: 0.0521628	total: 14m 23s	remaining: 5m 39s
359:	learn: 0.0520638	total: 14m 26s	remaining: 5m 36s
360:	learn: 0.0519565	total: 14m 28s	remaining: 5m 34s
361:	learn: 0.0518106	total: 14m 30s	remaining: 5m 32s
362:	learn: 0.0517033	total: 14m 33s	remaining: 5m 29s
363:	learn: 0.0515867	total: 14m 35s	remaining: 5m 27s
364:	learn: 0.0514802

497:	learn: 0.0419658	total: 20m 3s	remaining: 4.83s
498:	learn: 0.0419107	total: 20m 5s	remaining: 2.42s
499:	learn: 0.0418626	total: 20m 8s	remaining: 0us
Iteration No: 16 ended. Search finished for the next optimal point.
Time taken: 1211.3922
Function value obtained: -0.9602
Current minimum: -0.9701
Iteration No: 17 started. Searching for the next optimal point.
0:	learn: 2.1446822	total: 2.52s	remaining: 4m 9s
1:	learn: 1.9195104	total: 5s	remaining: 4m 5s
2:	learn: 1.7504104	total: 7.55s	remaining: 4m 4s
3:	learn: 1.6178890	total: 10s	remaining: 4m 1s
4:	learn: 1.5073322	total: 12.6s	remaining: 3m 58s
5:	learn: 1.4109522	total: 15s	remaining: 3m 55s
6:	learn: 1.3276949	total: 17.4s	remaining: 3m 51s
7:	learn: 1.2556327	total: 19.8s	remaining: 3m 47s
8:	learn: 1.1894217	total: 22.2s	remaining: 3m 44s
9:	learn: 1.1305226	total: 24.7s	remaining: 3m 42s
10:	learn: 1.0772829	total: 27.1s	remaining: 3m 38s
11:	learn: 1.0290179	total: 29.5s	remaining: 3m 36s
12:	learn: 0.9851604	total: 

46:	learn: 0.0956812	total: 1m 52s	remaining: 18m 5s
47:	learn: 0.0940063	total: 1m 54s	remaining: 18m 2s
48:	learn: 0.0926233	total: 1m 57s	remaining: 18m
49:	learn: 0.0911034	total: 1m 59s	remaining: 17m 58s
50:	learn: 0.0894140	total: 2m 2s	remaining: 17m 57s
51:	learn: 0.0878135	total: 2m 4s	remaining: 17m 56s
52:	learn: 0.0866267	total: 2m 7s	remaining: 17m 53s
53:	learn: 0.0855449	total: 2m 9s	remaining: 17m 51s
54:	learn: 0.0843515	total: 2m 12s	remaining: 17m 48s
55:	learn: 0.0829294	total: 2m 14s	remaining: 17m 46s
56:	learn: 0.0810018	total: 2m 16s	remaining: 17m 44s
57:	learn: 0.0802512	total: 2m 19s	remaining: 17m 41s
58:	learn: 0.0789020	total: 2m 21s	remaining: 17m 39s
59:	learn: 0.0781434	total: 2m 24s	remaining: 17m 36s
60:	learn: 0.0768982	total: 2m 26s	remaining: 17m 34s
61:	learn: 0.0757416	total: 2m 28s	remaining: 17m 31s
62:	learn: 0.0751816	total: 2m 31s	remaining: 17m 29s
63:	learn: 0.0743317	total: 2m 33s	remaining: 17m 27s
64:	learn: 0.0731163	total: 2m 36s	rem

198:	learn: 0.0342330	total: 7m 58s	remaining: 12m 4s
199:	learn: 0.0341534	total: 8m 1s	remaining: 12m 2s
200:	learn: 0.0340703	total: 8m 3s	remaining: 11m 59s
201:	learn: 0.0339645	total: 8m 6s	remaining: 11m 57s
202:	learn: 0.0338863	total: 8m 8s	remaining: 11m 54s
203:	learn: 0.0337664	total: 8m 11s	remaining: 11m 52s
204:	learn: 0.0336745	total: 8m 13s	remaining: 11m 50s
205:	learn: 0.0335886	total: 8m 15s	remaining: 11m 47s
206:	learn: 0.0334218	total: 8m 18s	remaining: 11m 45s
207:	learn: 0.0333435	total: 8m 20s	remaining: 11m 42s
208:	learn: 0.0332594	total: 8m 22s	remaining: 11m 40s
209:	learn: 0.0331971	total: 8m 25s	remaining: 11m 37s
210:	learn: 0.0331306	total: 8m 27s	remaining: 11m 35s
211:	learn: 0.0330760	total: 8m 30s	remaining: 11m 32s
212:	learn: 0.0329824	total: 8m 32s	remaining: 11m 30s
213:	learn: 0.0328808	total: 8m 35s	remaining: 11m 28s
214:	learn: 0.0327527	total: 8m 37s	remaining: 11m 25s
215:	learn: 0.0326989	total: 8m 39s	remaining: 11m 23s
216:	learn: 0.03

349:	learn: 0.0244424	total: 14m 6s	remaining: 6m 2s
350:	learn: 0.0244051	total: 14m 9s	remaining: 6m
351:	learn: 0.0243728	total: 14m 11s	remaining: 5m 58s
352:	learn: 0.0243370	total: 14m 14s	remaining: 5m 55s
353:	learn: 0.0243167	total: 14m 16s	remaining: 5m 53s
354:	learn: 0.0242962	total: 14m 19s	remaining: 5m 51s
355:	learn: 0.0242614	total: 14m 21s	remaining: 5m 48s
356:	learn: 0.0242253	total: 14m 24s	remaining: 5m 46s
357:	learn: 0.0241840	total: 14m 26s	remaining: 5m 43s
358:	learn: 0.0241618	total: 14m 29s	remaining: 5m 41s
359:	learn: 0.0241270	total: 14m 31s	remaining: 5m 39s
360:	learn: 0.0240574	total: 14m 34s	remaining: 5m 36s
361:	learn: 0.0240209	total: 14m 37s	remaining: 5m 34s
362:	learn: 0.0239811	total: 14m 39s	remaining: 5m 31s
363:	learn: 0.0239450	total: 14m 42s	remaining: 5m 29s
364:	learn: 0.0239025	total: 14m 45s	remaining: 5m 27s
365:	learn: 0.0238558	total: 14m 48s	remaining: 5m 25s
366:	learn: 0.0238141	total: 14m 50s	remaining: 5m 22s
367:	learn: 0.023

Iteration No: 18 ended. Search finished for the next optimal point.
Time taken: 1220.7907
Function value obtained: -0.9685
Current minimum: -0.9701
Iteration No: 19 started. Searching for the next optimal point.
0:	learn: 1.6940780	total: 2.61s	remaining: 21m 42s
1:	learn: 1.3974375	total: 5.17s	remaining: 21m 26s
2:	learn: 1.2040400	total: 7.68s	remaining: 21m 12s
3:	learn: 1.0625720	total: 10.2s	remaining: 21m 3s
4:	learn: 0.9475675	total: 12.7s	remaining: 20m 56s
5:	learn: 0.8529455	total: 15.1s	remaining: 20m 40s
6:	learn: 0.7795741	total: 17.5s	remaining: 20m 30s
7:	learn: 0.7107288	total: 20s	remaining: 20m 31s
8:	learn: 0.6524941	total: 22.4s	remaining: 20m 21s
9:	learn: 0.6024751	total: 24.7s	remaining: 20m 12s
10:	learn: 0.5614552	total: 27.2s	remaining: 20m 7s
11:	learn: 0.5242248	total: 29.6s	remaining: 20m 3s
12:	learn: 0.4873179	total: 32s	remaining: 20m
13:	learn: 0.4559641	total: 34.4s	remaining: 19m 55s
14:	learn: 0.4299825	total: 36.8s	remaining: 19m 49s
15:	learn: 0.4

149:	learn: 0.0463958	total: 6m 7s	remaining: 14m 18s
150:	learn: 0.0461976	total: 6m 10s	remaining: 14m 15s
151:	learn: 0.0459290	total: 6m 12s	remaining: 14m 13s
152:	learn: 0.0455750	total: 6m 15s	remaining: 14m 10s
153:	learn: 0.0453483	total: 6m 17s	remaining: 14m 8s
154:	learn: 0.0450489	total: 6m 20s	remaining: 14m 5s
155:	learn: 0.0448040	total: 6m 22s	remaining: 14m 3s
156:	learn: 0.0444626	total: 6m 25s	remaining: 14m 1s
157:	learn: 0.0442073	total: 6m 27s	remaining: 13m 58s
158:	learn: 0.0440096	total: 6m 29s	remaining: 13m 55s
159:	learn: 0.0437997	total: 6m 32s	remaining: 13m 53s
160:	learn: 0.0435064	total: 6m 34s	remaining: 13m 50s
161:	learn: 0.0432461	total: 6m 36s	remaining: 13m 47s
162:	learn: 0.0430549	total: 6m 39s	remaining: 13m 45s
163:	learn: 0.0428961	total: 6m 41s	remaining: 13m 42s
164:	learn: 0.0426338	total: 6m 44s	remaining: 13m 40s
165:	learn: 0.0424547	total: 6m 46s	remaining: 13m 37s
166:	learn: 0.0422657	total: 6m 48s	remaining: 13m 35s
167:	learn: 0.0

299:	learn: 0.0262215	total: 12m 10s	remaining: 8m 6s
300:	learn: 0.0261123	total: 12m 12s	remaining: 8m 4s
301:	learn: 0.0260303	total: 12m 15s	remaining: 8m 2s
302:	learn: 0.0259484	total: 12m 17s	remaining: 7m 59s
303:	learn: 0.0259139	total: 12m 20s	remaining: 7m 57s
304:	learn: 0.0258584	total: 12m 22s	remaining: 7m 54s
305:	learn: 0.0257906	total: 12m 25s	remaining: 7m 52s
306:	learn: 0.0257298	total: 12m 27s	remaining: 7m 49s
307:	learn: 0.0256407	total: 12m 29s	remaining: 7m 47s
308:	learn: 0.0255741	total: 12m 32s	remaining: 7m 45s
309:	learn: 0.0255161	total: 12m 34s	remaining: 7m 42s
310:	learn: 0.0254434	total: 12m 37s	remaining: 7m 40s
311:	learn: 0.0253821	total: 12m 39s	remaining: 7m 37s
312:	learn: 0.0253232	total: 12m 41s	remaining: 7m 35s
313:	learn: 0.0252507	total: 12m 44s	remaining: 7m 32s
314:	learn: 0.0251853	total: 12m 46s	remaining: 7m 30s
315:	learn: 0.0251287	total: 12m 49s	remaining: 7m 27s
316:	learn: 0.0250569	total: 12m 51s	remaining: 7m 25s
317:	learn: 0

450:	learn: 0.0185619	total: 18m 15s	remaining: 1m 59s
451:	learn: 0.0185218	total: 18m 18s	remaining: 1m 56s
452:	learn: 0.0184871	total: 18m 20s	remaining: 1m 54s
453:	learn: 0.0184427	total: 18m 22s	remaining: 1m 51s
454:	learn: 0.0184128	total: 18m 25s	remaining: 1m 49s
455:	learn: 0.0183760	total: 18m 27s	remaining: 1m 46s
456:	learn: 0.0183438	total: 18m 30s	remaining: 1m 44s
457:	learn: 0.0183168	total: 18m 32s	remaining: 1m 42s
458:	learn: 0.0182844	total: 18m 35s	remaining: 1m 39s
459:	learn: 0.0182162	total: 18m 37s	remaining: 1m 37s
460:	learn: 0.0181783	total: 18m 40s	remaining: 1m 34s
461:	learn: 0.0181556	total: 18m 42s	remaining: 1m 32s
462:	learn: 0.0181027	total: 18m 44s	remaining: 1m 29s
463:	learn: 0.0180718	total: 18m 47s	remaining: 1m 27s
464:	learn: 0.0180437	total: 18m 49s	remaining: 1m 25s
465:	learn: 0.0180085	total: 18m 52s	remaining: 1m 22s
466:	learn: 0.0179561	total: 18m 54s	remaining: 1m 20s
467:	learn: 0.0179234	total: 18m 57s	remaining: 1m 17s
468:	learn

100:	learn: 0.3647763	total: 4m 5s	remaining: 16m 8s
101:	learn: 0.3610979	total: 4m 7s	remaining: 16m 6s
102:	learn: 0.3571840	total: 4m 10s	remaining: 16m 4s
103:	learn: 0.3536988	total: 4m 12s	remaining: 16m 1s
104:	learn: 0.3502255	total: 4m 14s	remaining: 15m 59s
105:	learn: 0.3470287	total: 4m 17s	remaining: 15m 56s
106:	learn: 0.3438089	total: 4m 19s	remaining: 15m 53s
107:	learn: 0.3406329	total: 4m 22s	remaining: 15m 51s
108:	learn: 0.3375746	total: 4m 24s	remaining: 15m 48s
109:	learn: 0.3343593	total: 4m 26s	remaining: 15m 45s
110:	learn: 0.3312551	total: 4m 29s	remaining: 15m 43s
111:	learn: 0.3280893	total: 4m 31s	remaining: 15m 40s
112:	learn: 0.3252536	total: 4m 33s	remaining: 15m 37s
113:	learn: 0.3220920	total: 4m 36s	remaining: 15m 35s
114:	learn: 0.3190816	total: 4m 38s	remaining: 15m 32s
115:	learn: 0.3161736	total: 4m 41s	remaining: 15m 30s
116:	learn: 0.3134918	total: 4m 43s	remaining: 15m 28s
117:	learn: 0.3106496	total: 4m 45s	remaining: 15m 25s
118:	learn: 0.30

251:	learn: 0.1366896	total: 10m 5s	remaining: 9m 55s
252:	learn: 0.1360885	total: 10m 7s	remaining: 9m 53s
253:	learn: 0.1355931	total: 10m 10s	remaining: 9m 51s
254:	learn: 0.1349978	total: 10m 12s	remaining: 9m 48s
255:	learn: 0.1344210	total: 10m 15s	remaining: 9m 46s
256:	learn: 0.1338954	total: 10m 17s	remaining: 9m 43s
257:	learn: 0.1333650	total: 10m 19s	remaining: 9m 41s
258:	learn: 0.1327802	total: 10m 22s	remaining: 9m 39s
259:	learn: 0.1321948	total: 10m 24s	remaining: 9m 36s
260:	learn: 0.1318155	total: 10m 26s	remaining: 9m 34s
261:	learn: 0.1311931	total: 10m 29s	remaining: 9m 31s
262:	learn: 0.1306413	total: 10m 31s	remaining: 9m 29s
263:	learn: 0.1300698	total: 10m 34s	remaining: 9m 27s
264:	learn: 0.1295369	total: 10m 36s	remaining: 9m 24s
265:	learn: 0.1289967	total: 10m 39s	remaining: 9m 22s
266:	learn: 0.1284352	total: 10m 42s	remaining: 9m 20s
267:	learn: 0.1280090	total: 10m 44s	remaining: 9m 18s
268:	learn: 0.1275003	total: 10m 47s	remaining: 9m 16s
269:	learn: 

401:	learn: 0.0858938	total: 16m 11s	remaining: 3m 56s
402:	learn: 0.0856942	total: 16m 14s	remaining: 3m 54s
403:	learn: 0.0855229	total: 16m 16s	remaining: 3m 52s
404:	learn: 0.0853308	total: 16m 19s	remaining: 3m 49s
405:	learn: 0.0850926	total: 16m 21s	remaining: 3m 47s
406:	learn: 0.0848865	total: 16m 24s	remaining: 3m 44s
407:	learn: 0.0846865	total: 16m 26s	remaining: 3m 42s
408:	learn: 0.0845325	total: 16m 28s	remaining: 3m 40s
409:	learn: 0.0843680	total: 16m 31s	remaining: 3m 37s
410:	learn: 0.0841551	total: 16m 33s	remaining: 3m 35s
411:	learn: 0.0839622	total: 16m 36s	remaining: 3m 32s
412:	learn: 0.0837890	total: 16m 38s	remaining: 3m 30s
413:	learn: 0.0836242	total: 16m 40s	remaining: 3m 27s
414:	learn: 0.0833829	total: 16m 43s	remaining: 3m 25s
415:	learn: 0.0832065	total: 16m 45s	remaining: 3m 23s
416:	learn: 0.0830640	total: 16m 48s	remaining: 3m 20s
417:	learn: 0.0829170	total: 16m 50s	remaining: 3m 18s
418:	learn: 0.0827473	total: 16m 52s	remaining: 3m 15s
419:	learn

In [39]:
catboost_model = CatBoostClassifier(iterations=500,
                           learning_rate=0.16,
                           depth=10,
                           l2_leaf_reg=6,
                           custom_metric=['TotalF1:average=Macro']
                           )

catboost_model.fit(X_train_filled, y_train_prepared, verbose=False)

score_7= catboost_model.score(X_test_filled, y_test_prepared)

print(f"Score: {score_7}")

Score: 0.9701270667570354


As conclusion, we can say that filling the missing values with average values depending on class label gave good result rising score on test from 77% to 91%. Model tuning rised the accuracy of prediction to 97%. <br>Unfortunately, oversampling did not increase the score on test. Methods which being used also are 1) CatBoost and Virgin Data, 2) CatBoost + normalized data and replaced missing values(-1), 3) removing features that contains more than 50% of missing values and 4) Stochastic KNN Imputer of gaps. 